extension/bootstrap.js

```diff
@@ -91,25 +169,12 @@ class CopyController {
     if (cmd === "cmd_copy") {
       studyUtils.telemetry({ event: "copy" });
       const shareButton = this.browserWindow.shareButton;
-      if (shareButton !== null && // the button exists
-          shareButton.getAttribute("disabled") !== "true" && // the page we are on can be shared
-          shareButton.getAttribute("cui-areatype") === "toolbar" && // the button is in the toolbar
-          shareButton.getAttribute("overflowedItem") !== "true") { // but not in the overflow menu
-        // check to see if we should call a treatment at all
-        const numberOfTimeShown = Preferences.get("extensions.sharebuttonstudy.counter", 0);
-        if (numberOfTimeShown < MAX_TIMES_TO_SHOW) {
-          Preferences.set("extensions.sharebuttonstudy.counter", numberOfTimeShown + 1);
-
-          if (this.treatment === "ALL") {
-            Object.keys(TREATMENTS).forEach((key, index) => {
-              if (Object.prototype.hasOwnProperty.call(TREATMENTS, key)) {
-                TREATMENTS[key](this.browserWindow, shareButton);
-              }
-            });
-          } else if (this.treatment in TREATMENTS.keys()) {
-            TREATMENTS[this.treatment](this.browserWindow, shareButton);
-          }
-        }
+      // check to see if we should call a treatment at all
+      const numberOfTimeShown = Preferences.get(COUNTER_PREF, 0);
+      if (numberOfTimeShown < MAX_TIMES_TO_SHOW &&
+          Object.keys(TREATMENTS).includes(this.treatment)) {
```

Comment:
This should be false if either the treatment has been shown >= `MAX_TIMES_TO_SHOW` or if the treatment is not valid.